### PR TITLE
TF parser support for ternary conditions

### PR DESCRIPTION
**Pull request by robeden**
This introduces parser support for ternary (`${<condition> ? <then> : <else>}`) operations. If the `condition` portion can resolve to `true` or `false`, then the `then` or `else` portion is replaced accordingly. Conditions are extracted as proper variables to be resolved, so things like `var.foo` will be handled, if those can be resolved.

See the "ternary_793" and "ternaries" parser scenarios for a full test case (from #793).

Note that one case that doesn't currently work is mapping boolean string values:
```
foo = "true" ? "yes" : "no"
```
There is a bug with the HCL2 parser we use which currently blocks that functionality (see `test_hcl2_load_assumptions.py`'s `test_weird_ternary_string_clipping` case).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

Fixes: #793

### CODE REVIEW

File: tests/terraform/parser/resources/parser_scenarios/ternary_793/main.tf [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+resource "aws_instance" "foo" {
+  ami           = "ami-005e54dee72cc1d00" # us-west-2
+  instance_type = "t2.micro"
+
+  root_block_device {
+    encrypted = true
+  }
+
+  metadata_options {
+    http_tokens = (var.metadata_http_tokens_required) ? "required" : "optional"
```
Comment (nimrodkor) [inline tests/terraform/parser/resources/parser_scenarios/ternary_793/main.tf:16]: @robeden Can we test multi-line ternary? i.e.:
```terraform
 http_tokens = (var.metadata_http_tokens_required) ? 
    "required" : "optional"
```

**Review by nimrodkor — COMMENTED** [review]
Looks really good @robeden !
One test case I'd like to make sure works

File: tests/terraform/parser/resources/parser_scenarios/ternary_793/main.tf (hunk 1 above)
Comment (robeden) [reply tests/terraform/parser/resources/parser_scenarios/ternary_793/main.tf:16]: Looks like the HCL2 parser dies on that:

> Unexpected token Token(__ANON_0, '\\n') at line 10, column 35.
> Expected one of: 
>  * LSQB
>  * LBRACE
>  * LPAR
>  * NULL
>  * __ANON_11
>  * DECIMAL
>  * FALSE
>  * __ANON_3
>  * __ANON_10
>  * STRING_LIT
>  * MINUS
>  * TRUE
>  * BANG

Comment (robeden) [reply tests/terraform/parser/resources/parser_scenarios/ternary_793/main.tf:16]: I added this to the "ternaries" test case:
```
  local_true = true
  multiline = (local.local_true) ?
    "correct" : "wrong"
```

(Note: local evals inside locals isn't valid in terraform, but our parser would handle it... if the HCL2 parser would.)

Comment (nimrodkor) [reply tests/terraform/parser/resources/parser_scenarios/ternary_793/main.tf:16]: @robeden Is it on a commit you haven't pushed yet?

Comment (robeden) [reply tests/terraform/parser/resources/parser_scenarios/ternary_793/main.tf:16]: Yes, sorry. Pushed now.

File: tests/terraform/parser/resources/parser_scenarios/ternaries/main.tf [hunk 2]
```diff
@@ -6,6 +6,11 @@ locals {
   bool_true  = true ? "correct" : "wrong"
   bool_false = false ? "wrong" : "correct"
 
+//  local_true = true
+  // TODO: HCL2 parser doesn't like the following line
+//  multiline = (local.local_true) ?
+//    "correct" : "wrong"
```
Comment (nimrodkor) [inline tests/terraform/parser/resources/parser_scenarios/ternaries/main.tf:12]: @robeden I see it is commented out, so I'm guessing we don't support it? If not I'd like to have an issue on it

Comment (robeden) [reply tests/terraform/parser/resources/parser_scenarios/ternaries/main.tf:12]: Filed #864. Note that #863 is also relevant to this PR.

**Review by nimrodkor — APPROVED** [review]

**Pull request merged**